core/src/main/java/org/elasticsearch/index/query/QueryBuilder.java

```diff
@@ -30,6 +36,57 @@
 
 public interface QueryBuilder<QB extends QueryBuilder> extends NamedWriteable<QB>, ToXContent {
 
+    public static enum Operator implements Writeable {
```

Comment:
can we move Operator out to his own class on this same package? I think you missed my previous comment about this.